### PR TITLE
[ios][android][qt] Add application version to Platform

### PR DESCRIPTION
**Pull request by rtsisyk**
Signed-off-by: Roman Tsisyk <roman@tsisyk.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -154,7 +154,7 @@ designer_version.h
 *.sw?
```
Comment (rtsisyk) [inline .gitignore:154]: This patch solves the following issues:

1. For iOS `build_version.hpp` wasn't generated at all because Xcode doesn't use cmake
2. For Android `build_version::kName` didn't contain flavorName and buildType.

This old/new implementation from 1y old patch fixes both problems by adding two methods to Platform. The version code a.k.a. intVersion has been reduced to yy_MM_dd. It matches dataVersion (also int32) and it should be more than enough for any conditionals in the code.

File: tools/unix/generate_version.sh [hunk 2]
```diff
@@ -1,75 +0,0 @@
-#!/bin/bash
```
Comment (rtsisyk) [inline tools/unix/generate_version.sh:1]: This is an ancient version of `version.sh` script. It is not used at all.

**Review by vng — APPROVED** [review]

File: platform/platform_ios.mm [hunk 3]
```diff
@@ -157,6 +157,23 @@
...
+}
+
+int32_t Platform::IntVersion() const
+{
+  NSString * version = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"CFBundleShortVersionString"];
+  int year = 0;
+  int month = 0;
+  int day = 0;
+  int rc = sscanf(version.UTF8String, "%d.%d.%d", &year, &month, &day);
+  CHECK_EQUAL(rc, 3, ("Failed t parse version"));
```
Comment (biodranik) [inline platform/platform_ios.mm:173]: to

File: platform/platform_ios.mm [hunk 4]
```diff
@@ -157,6 +157,23 @@
...
+  return (int32_t)(year - 2000) * 10000 + month * 100 + day;
```
Comment (biodranik) [inline platform/platform_ios.mm:174]: Add ASSERTs for valid y m d values

File: tools/unix/version.sh [hunk 5]
```diff
@@ -42,8 +42,17 @@ function android_code {
   echo "${cutYear//./}$(printf %02d "$COUNT")"
 }
 
-function git_hash {
-  git describe --match="" --always --abbrev=8 --dirty
```
Comment (biodranik) [inline tools/unix/version.sh:46]: Why did you remove it? Please leave it in the string version on all platforms, it will help us to debug or to distinguish between someone's else "custom" versions.

File: tools/unix/version.sh [hunk 6]
```diff
@@ -42,8 +42,17 @@ function android_code {
...
+  # yy_MM_dd
+  # yy - year
+  # MM - month
+  # dd - day
+  local cutYear=${DATE:2}
+  echo "${cutYear//./}"
+}
+
+function qt_version {
+  echo "$DATE-$COUNT ($(git describe --match="" --always --abbrev=8 --dirty))"
```
Comment (biodranik) [inline tools/unix/version.sh:55]: ???

File: tools/unix/version.sh [hunk 7]
```diff
@@ -42,8 +42,17 @@ function android_code {
...
 
-function git_hash {
-  git describe --match="" --always --abbrev=8 --dirty
+function qt_int_version {
+  # yy_MM_dd
+  # yy - year
+  # MM - month
+  # dd - day
+  local cutYear=${DATE:2}
+  echo "${cutYear//./}"
```
Comment (biodranik) [inline tools/unix/version.sh:51]: Why do you need a custom version for Qt? Let's reuse the same as for other platforms.

**Review by biodranik — CHANGES_REQUESTED** [review]

File: tools/unix/version.sh (hunk 5 above)
Comment (rtsisyk) [reply tools/unix/version.sh:46]: Git has been removed from version string on Android because it created the chicken or the egg problem (context: https://github.com/organicmaps/organicmaps/issues/825). 

- Android: `2022.08.19-7-FDroid` - generated by Gradle by using data frversion.sh
- iOS: `2022.08.19-7` - generated by Xcode, don't have a place for githash
- Qt: `2022.08.19-7 (githash)`

File: tools/unix/version.sh (hunk 7 above)
Comment (rtsisyk) [reply tools/unix/version.sh:51]: Actually, Qt version doesn't really matter, but I tried to preserve format added before my patch.

File: platform/platform_ios.mm (hunk 3 above)
Comment (rtsisyk) [reply platform/platform_ios.mm]: Fixed.

File: platform/platform_ios.mm (hunk 4 above)
Comment (rtsisyk) [reply platform/platform_ios.mm:177]: Added CHECKs.

File: tools/unix/version.sh (hunk 7 above)
Comment (vng) [reply tools/unix/version.sh:51]: Int app version should be the same for all platform like 220816. It will be used in for dynamic data updates.

**Review by vng — APPROVED** [review]

File: tools/unix/version.sh (hunk 7 above)
Comment (rtsisyk) [reply tools/unix/version.sh:51]: Yes, of course, intVersion() is the same for all platforms - 220816. I don't think that we need one more section here (220816 instead of 22081601).

File: tools/unix/version.sh (hunk 6 above)
Comment (biodranik) [reply tools/unix/version.sh:55]: local GIT_HASH=$(git describe --match="" --always --abbrev=8 --dirty)

**Review by biodranik — APPROVED** [review]

File: tools/unix/version.sh (hunk 6 above)
Comment (rtsisyk) [reply tools/unix/version.sh]: I have changed the format for Qt version to be consistent with Android/iOS:

- `2022.08.21-3-ad79d64e-Darwin` (instead of `2022.08.21-3 (ad79d64e)`)
- `2022.08.19-7-FDroid`
- `2022.08.19-7`
